sdk/core/scheduler/common.h

```diff
@@ -247,6 +247,10 @@ namespace
 		HeapObject(struct SObjStruct *heapCapability, T *allocatedObject)
 		  : pointer(allocatedObject, heapCapability)
 		{
+			if (__builtin_cheri_tag_get(allocatedObject) == 0)
```

Comment:
This built in should return a bool, so comparing it against zero is confusing.

Reply:
Sounds good. I just replicated what I saw there: https://github.com/microsoft/cheriot-rtos/blob/main/tests/allocator-test.cc#L87

Should I fix this as well?

Reply:
Yes please!